backends/p4tools/modules/testgen/options.cpp

```diff
@@ -121,27 +173,88 @@ TestgenOptions::TestgenOptions()
     registerOption(
         "--exploration-strategy", "explorationStrategy",
         [this](const char* arg) {
-            explorationStrategy = arg;
+            explorationStrategy = cstring(arg).toUpper();
+            if (SUPPORTED_EXPLORATION_STRATEGIES.count(explorationStrategy) == 0) {
+                ::error(
+                    "Exploration strategy %1% not supported. Supported exploration strategies are "
+                    "%2%.",
+                    explorationStrategy,
+                    Utils::containerToString(SUPPORTED_EXPLORATION_STRATEGIES));
+                return false;
+            }
             return true;
         },
         "Selects a specific exploration strategy for test generation. Options are: "
-        "randomAccessStack, linearEnumeration, maxCoverage. Defaults to incrementalStack.");
+        "INCREMENTAL_STACK, RANDOM_ACCESS_STACK, LINEAR_ENUMERATION, MAX_COVERAGE, and "
+        "RANDOM_ACCESS_MAX_COVERAGE. Defaults to "
+        "INCREMENTAL_STACK.");
+
+    registerOption(
+        "--pop-level", "popLevel",
+        [this](const char* arg) {
+            int64_t popLevelTmp = 0;
+            try {
+                // Unfortunately, we can not use std::stoul because negative inputs are okay
+                // according to the C++ standard.
+                popLevelTmp = std::stoll(arg);
+                if (popLevelTmp < 0) {
+                    throw std::invalid_argument("Negative input.");
```

Comment:
If we're throwing invalid arg here, would it make sense to use the right ranges already? i.e. `popLevel` should not be negative, but it also needs to be greater than 1. 

Maybe an elegant improvement would be creating a function in `Utils`, i.e. `Utils::positiveGreaterThan(BOUND)` ? Then it could be reused in `popLevel`, `linearEnumeration` and `saddlePoint`.

Reply:
Fixed. Regarding the second point, I considered this but we still have to catch the exception and use the appropriate type so the gain is relatively small.